Add version switcher, favicon, and context-dependent logo

nwlandry opened the pull request:
Fixes #584 and #585.

This PR
* Adds a favicon
* Creates a dark and light version of the logo
* Adds a version switcher

maximelucas commented:
Amazing Nich!! Favicon is a sweet plus.

Two things: 
1. On my laptop the menu is now looking a bit strange because on two lines
2. the light theme logo doesn't have a margin anymore so it's stuck to the "new release" button (dark theme logo doesn't have that problem)

![image](https://github.com/user-attachments/assets/61a8c2ef-e072-42b2-ac5a-e1ab378fbaf8)

I'm not fully sure how to solve 1., I think we need to find a way to control the width of the[ 3 navbars/zones (left, middle, right)](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/layout.html)? Can the version switcher be made less wide by changing the default text from "choose version" to "vesion" or simply "stable" by default?

nwlandry commented:
In response to the review I
* reverted the logo back
* adjusted the spacing in the navbar
* Added a call to the Github API to get the date of the latest release

I believe I addressed all your comments, @maximelucas!

maximelucas commented on docs/source/conf.py:
```diff
@@ -115,9 +118,18 @@
+
+if r.ok:
+    release_date = (
+        datetime.fromisoformat(r.json()["published_at"]).date().strftime("%B %d, %Y")
```
```suggestion
        datetime.fromisoformat(r.json()["published_at"]).date().strftime("%b %d, %Y")
```

maximelucas replied:
just short month version for space

maximelucas commented on docs/source/conf.py:
```diff
@@ -184,6 +201,13 @@
             "icon": "fa-brands fa-mastodon",  # Font Awesome icon
         },
     ],
+    "header_links_before_dropdown": 4,
```
Anyway of having 5 and keeping the Gallery visible (not in dropdown)?

nwlandry replied:
The issue was that I wanted everything on one line even when the user wasn't in full-screen, which is basically impossible unless the navbar is slimmed down. So I will keep it like it is for now, and then change it later if need be.

maximelucas commented:
You're on fire, the automatic date retrieval is great!!
Two tiny comments but I'll let you choose, approving already.

nwlandry commented:
Thanks for the review!
